src/recon_buildblock/IterativeReconstruction.cxx

```diff
@@ -397,16 +397,16 @@ reconstruct()
 {
   this->start_timers();
 
-  this->target_data_sptr.reset(this->get_initial_data_ptr());
-  if (this->set_up(this->target_data_sptr) == Succeeded::no)
+  shared_ptr<TargetT > target_data_sptr(this->get_initial_data_ptr());
```

Comment:
Can you explain this to me? It looks incredibly dangerous to change to using from a class member to a local variable. Might be better, but surely breaks backwards compatibility in a silent way

Reply:
I believe that all this comes from the scatter branch. I had the need to be able to reconstruct / get the reconstructed image / process / put back, all without saving the image at any point but the end. 
Looking this out of context indeed looks dangerous. But throughout PR we keep removing this functionality. It might look out of place, but we had said add the more reconstruction related stuff in this branch.

Reply:
edit: I am not sure if the process chain can be achieved with a global variable

Reply:
I'll veto this change I'm afraid. It is too dangerous. If we need it for scatter, let's discuss it there. But I really can't see why this change would help you avoiding to save the image. A better way to do that is to reset the OutputFileFormat (or set it to a OutputFileFormat-type that doesn't save, a trick that was used before by Oliver Nix).

`this->target_data_sptr` is not a global variable of course, it is a member of the object.  I fail to see why a variable local to the function would allow you to do more than a member-variable.

Reply:
You are right. I looked a bit deeper into this. So if you look here: https://github.com/UCL/STIR/commit/3d0aa82ca0559de0f4846322d3f809d70643e99e#diff-275b51d3aa1eb9f308fa984415d11673
I had this fixed back on July 2016 and the method to reconstruct without output has been accepted by you. But git brought this back for some reason.